### DISCUSSION TITLE
Exclude project for reporting

### DISCUSSION THREAD

**Discussion by Solidus-Prime** [kimai/kimai#2505] [General]
Is there a way to exclude a particular project from the reports?

We have a project we'd like to track time on for informational purposes, but we would like it to not show up on our weekly payroll reports. Is this possible?

**Comment by Solidus-Prime** [comment]
I am only half-proficient with PHP, so I am winging it as best I can :)

It looks like the report is put together using the 'ReportingController'? If I add a custom field to the project I want to "ignore", is it possible to add a line to the controller that tells it to ignore this custom field in it's calculations, or 'get' the amount of any projects with that custom field, and then subtract that same amount from the total?

Something like this?

if ($project_ID == 'PROJECT_ID_TO_EXLUDE') {
continue;
}

**Reply by kevinpapst** [reply]
What is a "weekly payroll report" ?

**Reply by Solidus-Prime** [reply]
Apologies - We use the Reporting widget in general for our "Weekly payroll report". Mainly the 'weekly view for one user".

In our work environment the employees clock in at the beginning of the day, and clock out at the end of the day. At the end of the week, they are paid based on the sum of these hours. 

Throughout the day, employees clock into and out of various projects, each with it's own activities. This is to accurately track time spent per project, based on activity.

 So John comes in at 8 AM, and clocks in to the project "Payroll". Then he clocks into "Project 1 - Activity A" for 15 mins. When it's complete, he punches out of that, and punches into "Project 1 - Activity B" for 2 hours. Then "Project 2 - Activity D", and so on and so forth throughout the day. When he's ready to leave for the day, he punches out of "Payroll".

This lets us track the time he spent on each project, as well as the time he spent in the building. If we don't use the broad "Payroll" project the way we do, the clock doesn't accurately reflect the time they are in the building because they have time between projects when they are doing things like cleaning, preparing, or even just walking from one place to another. The building is pretty big in some places, and sometimes it takes an employee 10 minutes just to get to the next machine they need to use.

If we could stop "Payroll" from being calculated into the report totals that would be great. If that would be too difficult removing it from the report entirely would suffice if they could still punch to it because I track the payroll project hours in a Crystal report on the side.

**Reply by Solidus-Prime** [reply]
> What is a "weekly payroll report" ?

Do you have any suggestions, by chance?

I was thinking I could add something to one of the reporting controllers, or somehow ignore "Payroll" in the duration calculator, but have not had any luck so far. I've been trying to brush up on my PHP skills in the meantime because I feel like this is probably a simple fix that I am missing because I'm not very skilled.

**Reply by kevinpapst** [reply]
There is no simple/satisfying answer. But obviously you can filter the data, if you customize the code. 
I don't know how without investing time. And I have no free time right now.
If you want me to think about a solution (for your company), I am available for paid freelance work.

**Reply by Solidus-Prime** [reply]
> There is no simple/satisfying answer. But obviously you can filter the data, if you customize the code.
> I don't know how without investing time. And I have no free time right now.
> If you want me to think about a solution (for your company), I am available for paid freelance work.

Ok. If you want to provide a time/cost estimate I will pass it along.
